src/vm.rs

```diff
@@ -552,7 +553,11 @@ impl<'a, V: Verifier, C: ContextObject> EbpfVm<'a, V, C> {
                 stopwatch_numerator: 0,
                 stopwatch_denominator: 0,
                 program_result: ProgramResult::Ok(0),
-                memory_mapping: MemoryMapping::new(regions, config)?,
+                memory_mapping: if let Some(cow_cb) = cow_cb {
```

Comment:
I think we should refactor the constructor to just accept a `MemoryMapping` here instead of all its parts. It needs to become more versatile anyway to support different ABIs.

Reply:
Ok I've done this now